src/main/java/bookstore/service/order/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+import bookstore.repository.order.OrderRepository;
+import bookstore.service.order.OrderService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    public List<OrderDto> getAllOrders(String email, Pageable pageable) {
+        return orderMapper.toDtos(orderRepository.findAllByUserEmail(email, pageable));
+    }
+
+    @Transactional
+    @Override
+    public OrderDto setOrderStatus(Long id, UpdateOrderDto updateOrderDto) {
+        Order order = orderRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can not find order by id: " + id));
+
+        order.setStatus(updateOrderDto.status());
+        return orderMapper.toDto(orderRepository.save(order));
+
+    }
+
+    @Transactional
+    @Override
+    public OrderDto saveOrder(String email, CreateOrderRequestDto requestDto) {
+        ShoppingCart userShoppingCart = shoppingCartRepository.findByUserEmail(email)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can not find user's shopping cart. User email: " + email));
+
+        Order userOrder = new Order(userShoppingCart);
+        userOrder.setShippingAddress(requestDto.shippingAddress());
+        userOrder.setTotal(userShoppingCart.getCartItems().stream()
+                .map(i -> i.getBook().getPrice())
+                .reduce(BigDecimal.ZERO, BigDecimal::add));
+        Set<OrderItem> orderItems = new HashSet<>();
+
+        for (CartItem item : userShoppingCart.getCartItems()) {
+            OrderItem orderItem = new OrderItem(item);
+            orderItems.add(orderItem);
+        }
+        userOrder.setOrderItems(orderItems);
+        return orderMapper.toDto(orderRepository.save(userOrder));
+    }
+
+    @Override
+    public List<OrderItemDto> getItems(Long id) {
+        Order order = orderRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(
+                "Can not find order by id:" + id));
+
+        return orderItemMapper.toDtos(order.getOrderItems());
+    }
+
+    @Override
+    public OrderItemDto getById(Long orderId, Long itemId) {
+        return orderItemMapper.toDto(orderItemRepository.getById(orderId, itemId).orElseThrow(
+                () -> new EntityNotFoundException(
+                        "Can not find order by id:" + itemId)));
+    }
```

Comment:
?
Your entity is uniquely identified by its id field. You don't need to accept orderId and pass it to the repository. Given item.id, you can get its order_id if you want to, it's a foreign key on the entity. Fix this service method and remove repo method